File: src/dataset/cpp/armstrong-numbers_exercism-Richard-m-j/claude/ClaudeSonnet4_armstrong_numbers_v1.cpp
#include "armstrong_numbers.h"
#include <cmath>
using namespace std;
namespace armstrong_numbers
{
    bool is_armstrong_number(int input)
    {
        if (input < 0) return false;
        if (input < 10) return true;
        
        int num = input;
        int numDigits = 0;
        
        // Count digits
        while (num > 0) {
            num /= 10;
            numDigits++;
        }
        
        num = input;
        long long sum = 0;
        
        // Calculate sum of powers
        while (num > 0) {
            int digit = num % 10;
            sum += static_cast<long long>(pow(digit, numDigits));
            num /= 10;
        }
        
        return sum == input;
    }
} // namespace armstrong_numbers

File: src/dataset/Java/armstrong-numbers_exercism-uzilan/claude/ClaudeSonnet4_ArmstrongNumbers_v1.java
class ArmstrongNumbers {

    boolean isArmstrongNumber(int numberToCheck) {
        if (numberToCheck < 0) return false;
        
        int original = numberToCheck;
        int numDigits = 0;
        int temp = numberToCheck;
        
        // Count digits
        do {
            numDigits++;
            temp /= 10;
        } while (temp > 0);
        
        // Calculate sum of powers
        long sum = 0;
        temp = numberToCheck;
        while (temp > 0) {
            int digit = temp % 10;
            sum += pow(digit, numDigits);
            temp /= 10;
        }
        
        return sum == original;
    }
    
    private long pow(int base, int exponent) {
        long result = 1;
        for (int i = 0; i < exponent; i++) {
            result *= base;
        }
        return result;
    }
}